build_rules.gradle

```diff
@@ -593,6 +593,149 @@ ext.applyJavaNature = {
+  String beamSdk = System.getProperty('beamSdk', 'java')
+
+  // Timeout (in seconds) after which PerfKit will stop executing the benchmark (and will fail).
+  String timeout = System.getProperty('itTimeout', '1200')
+
+  // Path to kubernetes configuration file.
+  String kubeconfig = System.getProperty('kubeconfig', System.getProperty('user.home') + '/.kube/config')
+
+  // Path to kubernetes executable.
+  String kubectl = System.getProperty('kubectl', 'kubectl')
+
+  // Paths to files with kubernetes infrastructure to setup before the test runs.
+  // PerfKit will have trouble reading 'null' path. It expects empty string if no scripts are expected.
+  String kubernetesScripts = System.getProperty('kubernetesScripts', '')
+
+  // Pipeline options to be used by the tested pipeline.
+  String integrationTestPipelineOptions = System.getProperty('integrationTestPipelineOptions')
+
+  // Path to file with 'dynamic' and 'static' pipeline options.
+  // that will be appended by perfkit to the test running command.
+  // PerfKit will have trouble reading 'null' path. It expects empty string if no config file is expected.
+  String optionsConfigFile = System.getProperty('beamITOptions', '')
+
+  // Fully qualified name of the test to be run, eg:
+  // 'org.apache.beam.sdks.java.io.jdbc.JdbcIOIT'.
+  String integrationTest = System.getProperty('integrationTest')
+
+  // Relative path to module where the test is, eg. 'sdks/java/io/jdbc.
+  String itModule = System.getProperty('itModule')
+
+  // Runner which will be used for running the tests.
+  String runner = System.getProperty('integrationTestRunner', 'direct')
+
+  // Any additional properties to be appended to benchmark execution command.
+  String extraProperties = System.getProperty('beamExtraProperties', '')
+}
+
+// When applied in a module's build.gradle file, this closure provides set of tasks
+// needed to run IOIT integration tests (manually, without perfkit).
+ext.enableJavaPerformanceTesting = {
+  println "enableJavaPerformanceTesting with ${it ? "$it" : "default configuration"} for project ${project.name}"
+
+  // Use the implicit it parameter of the closure to handle zero argument or one argument map calls.
+  // See: http://groovy-lang.org/closures.html#implicit-it
+  JavaPerformanceTestConfiguration configuration = it ? it as JavaPerformanceTestConfiguration : new JavaPerformanceTestConfiguration()
+
+  // Add runners needed to run integration tests on
+  task packageIntegrationTests(type: Jar) {
+      def runner = configuration.runner
+      dependencies {
+        if (runner.contains('dataflow')) {
+          testCompile project(path: ":runners:google-cloud-dataflow-java", configuration: 'shadowTest')
+        }
+
+        if (runner.contains('direct')) {
+          testCompile project(path: ":runners:direct-java", configuration: 'shadowTest')
+        }
+    }
+  }
+
+  // Task for running integration tests
+  task integrationTest(type: Test) {
```

Comment:
<!--new_thread; commit:a2d1efe7c481a850f0e744ee6d4372c885c695f8; resolved:0-->
Why do we have to define a new task for integration tests ? I think for Maven we used a standard task (verify). Is this a requirement for Gradle ?

Reply:
I wouldn't call this a requirement - rather a good gradle practice to handle things this way. 

For invoking the integration test maven used the Failsafe plugin. It was designed in a way to look for all test classes that name ends with "IT" and execute them. On the contrary unit tests were run using Surefire plugin. They both "connected" to mavens "verify" phase and this was a problem in maven: normally when someone typed `mvn verify` all plugins were fired. This is why we needed to "skipITs" or skip unit tests depending on which kind of tests we were going to execute and introduce profiles like "io-it" for this job: https://github.com/apache/beam/blob/b8cf07dc03643ded1d4e9d45caf4151316cf2c3e/sdks/java/io/pom.xml#L104

Gradle is built totally differently there and has no phases. It operates on independent tasks each representing "an atomic piece of work". This is IMO a perfect fit for our case: we define a Test type task called `integrationTest` or `performanceTest` and invoke it on demand, not worrying that unit tests will fire along with integration tests. Thanks to that we do not have to define a whole series of additional profiles. It's sufficient to have just two tasks defined in one place.